gemini/evaluation/evaluating_prompts_at_scale_with_gemini_batch_prediction_api.ipynb

```diff
@@ -0,0 +1,1011 @@
+    "\n",
+    "    get_status_url = (\n",
+    "        f\"https://us-central1-aiplatform.googleapis.com/v1/projects/{project_id}/\"\n",
+    "        f\"locations/us-central1/batchPredictionJobs/{job_id}\"\n",
+    "    )\n",
+    "\n",
+    "    # Get Authentication token.\n",
+    "    creds, _ = google.auth.default()\n",
+    "    auth_req = google.auth.transport.requests.Request()\n",
+    "    creds.refresh(auth_req)\n",
+    "    token = creds.token\n",
+    "\n",
+    "    headers = {\n",
+    "        \"Content-Type\": \"application/json\",\n",
+    "        \"Authorization\": f\"Bearer {token}\",\n",
+    "    }\n",
+    "\n",
+    "    job_state = last_job_state\n",
+    "    now = datetime.datetime.now()\n",
+    "    print(f\"Status {now}: {job_state}\")\n",
+    "\n",
+    "    while job_state not in STOP_STATES:\n",
+    "        state_response = requests.get(get_status_url, headers=headers)\n",
+    "        state_response_json = state_response.json()\n",
+    "        job_state = state_response_json[\"state\"]\n",
+    "        now = datetime.datetime.now()\n",
+    "        print(f\"Status {now}: {job_state}\")\n",
+    "        if job_state not in STOP_STATES:\n",
+    "            time.sleep(30)\n",
+    "\n",
+    "\n",
+    "check_job_state(project_id=PROJECT_ID, job_id=job_id, last_job_state=job_state)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "QTl8-Cbg7Ll5"
+   },
+   "source": [
+    "### List sample of text generation results from BigQuery"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "Y0jHGtZEUK88",
+    "outputId": "1c797b34-2113-4f08-ee6b-4278282fefd6"
+   },
+   "outputs": [],
+   "source": [
+    "def print_text_generation_results(text_generation_table):\n",
+    "    \"\"\"Lists text generation results from BigQuery.\"\"\"\n",
+    "\n",
+    "    job = bq_client.query(\n",
+    "        f\"\"\"\n",
+    "      SELECT gcs_uri, response, status, processed_time\n",
+    "      FROM {text_generation_table}\n",
+    "      LIMIT 5\"\"\"\n",
+    "    )\n",
+    "\n",
+    "    for row in job.result():\n",
+    "        json_row = json.loads(row[1])\n",
+    "        json_response = json_row[0][\"content\"][\"parts\"][0][\"text\"]\n",
+    "        print(json_response)\n",
+    "\n",
+    "\n",
+    "print_text_generation_results(text_generation_table=text_generation_table)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "01saqXPQbuIS"
+   },
+   "source": [
+    "## Analyze results in BigQuery and Looker Studio"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "VJvttnjyZxIs"
+   },
+   "source": [
+    "### Copy a Looker Studio dashboard to analyze results\n",
+    "\n",
+    "1. Make a copy of this [Looker Studio dashboard](https://lookerstudio.google.com/reporting/6dd5a7e8-b353-4dde-9bd8-72eb7b501559)\n",
```

Comment:
Non google.com accounts can't access this dashboard to copy it.

Reply:
Changed to a public dashboard